lib/test/createIdentity.js

```diff
@@ -0,0 +1,146 @@
+ * @param {Address} faucetAddress
+ * @param {PrivateKey} faucetPrivateKey
+ * @param {Address} address
+ * @return {Promise<string>}
+ */
+async function fundAddress(dapiClient, faucetAddress, faucetPrivateKey, address) {
+  const { items: inputs } = await dapiClient.getUTXO(faucetAddress.toString());
+
+  const transaction = new Transaction();
+
+  transaction.from(inputs.slice(-1)[0])
+    .to(address, 20000)
+    .change(faucetAddress)
+    .fee(668)
+    .sign(faucetPrivateKey);
+
+  const transactionId = await dapiClient.sendRawTransaction(transaction.serialize());
+
+  await dapiClient.generate(1);
+  await wait(5000);
+
+  return transactionId;
+}
+
+/**
+ *
+ * @param {DAPIClient} dapiClient
+ * @param {string} outPoint
+ * @param {number} type
+ * @param {PrivateKey} privateKey
+ * @param {IdentityPublicKey} identityPublicKey
+ * @return {Promise<IdentityCreateTransition>}
+ */
+async function applyIdentityCreateTransition(
+  dapiClient,
+  outPoint,
+  type,
+  privateKey,
+  identityPublicKey,
+) {
+  const identityCreateTransition = new IdentityCreateTransition({
+    protocolVersion: 0,
+    type: stateTransitionTypes.IDENTITY_CREATE,
+    lockedOutPoint: outPoint,
+    identityType: type,
+    publicKeys: [
+      identityPublicKey.toJSON(),
+    ],
+  });
+
+  identityCreateTransition.sign(identityPublicKey, privateKey);
+
+  await dapiClient.applyStateTransition(identityCreateTransition);
+
+  return identityCreateTransition;
+}
+
+/**
+ *
+ * @param {DAPIClient} dapiClient
+ * @param {PrivateKey} privateKey
+ * @param {number} type
+ * @return {Promise<Identity>}
+ */
+async function createIdentity(dapiClient, privateKey, type) {
+  // Create new keys
+  const publicKey = new PublicKey({
+    ...privateKey.toPublicKey().toObject(),
+    compressed: true,
+  });
+  const publicKeyBase = publicKey.toBuffer().toString('base64');
+  const address = Address
+    .fromPublicKey(publicKey, process.env.NETWORK === 'devnet' ? 'testnet' : process.env.NETWORK);
+
+  // eslint-disable-next-line no-underscore-dangle
+  const publicKeyHash = PublicKey.fromBuffer(Buffer.from(publicKeyBase, 'base64'))._getID();
+
+  const identityPublicKey = new IdentityPublicKey()
+    .setId(1)
+    .setType(IdentityPublicKey.TYPES.ECDSA_SECP256K1)
+    .setData(publicKeyBase);
+
+  // Found new address
+  const faucetPrivateKey = new PrivateKey(process.env.FAUCET_PRIVATE_KEY);
+  const faucetPublicKey = PublicKey.fromPrivateKey(faucetPrivateKey);
+  const faucetAddress = Address
+    .fromPublicKey(faucetPublicKey, process.env.NETWORK === 'devnet' ? 'testnet' : process.env.NETWORK);
+
+  await fundAddress(dapiClient, faucetAddress, faucetPrivateKey, address);
+
+  // Create Lock Transaction
+  const { items: [input] } = await dapiClient.getUTXO(address.toString());
+
+  const lockTransaction = new Transaction();
+
+  lockTransaction.from(input)
+    .addBurnOutput(10000, publicKeyHash)
+    .change(faucetAddress)
+    .fee(668)
+    .sign(privateKey);
+
```

Comment:
The same key pair is being used for the input address as for the identity. I'm guessing this was not done on purpose.

Reply:
Good catch! Thank you!

Reply:
@antouhou I forgot this part. So we agreed to use different private keys for a lock transaction and for identity what we going to create, right?

Reply:
Yep, as it's one of the best practices in cryptography - never reuse the same key for two different purposes